### PR TITLE
feat(core): programmatic setting of authz policy

### LINKED ISSUE #1768: Authz: enable setting policy programatically
Authz policy supports config as well as extending the default. Programmatic updates are desired.

### PR DESCRIPTION
**Pull request by jrschumacher**
Closes #1768

### Proposed Changes

* Enables programmatic setting of authz policy

### Checklist

- [x] I have added or updated unit tests
- [ ] I have added or updated integration tests (if appropriate)
- [x] I have added or updated documentation

### CODE REVIEW

**Review by jakedoublev — DISMISSED** [review]

File: service/pkg/server/options.go [hunk 1]
```diff
@@ -71,6 +72,21 @@ func WithAuthZDefaultPolicyExtension(policies [][]string) StartOptions {
 	}
 }
 
+// deprecated: Use WithAuthZPolicy
```
Comment (strantalis) [inline service/pkg/server/options.go:75]: Did you mean to put this on another option ?

Comment (jrschumacher) [reply service/pkg/server/options.go]: Yep good call!

File: service/pkg/server/options.go [hunk 2]
```diff
@@ -71,6 +73,21 @@ func WithAuthZDefaultPolicyExtension(policies [][]string) StartOptions {
 	}
 }
 
+// Deprecated: Use WithAuthZPolicy
```
Comment (strantalis) [inline service/pkg/server/options.go:76]:
```suggestion
```

File: service/pkg/server/options.go [hunk 3]
```diff
@@ -58,6 +59,7 @@ func WithPublicRoutes(routes []string) StartOptions {
 	}
 }
 
+// Deprecated: Use WithConfigKey
```
Comment (strantalis) [inline service/pkg/server/options.go:62]:
```suggestion
// Deprecated: Use WithAuthZPolicy
```

File: service/pkg/server/options.go [hunk 4]
```diff
@@ -12,11 +12,12 @@ type StartConfig struct {
 	WaitForShutdownSignal       bool
 	PublicRoutes                []string
 	authzDefaultPolicyExtension [][]string
+	authzPolicy                 string
 	extraCoreServices           []serviceregistry.IService
 	extraServices               []serviceregistry.IService
 }
 
-// Deprecated: Use WithConfigKey
+// WithConfigName option sets the configuration name.
```
Comment (strantalis) [inline service/pkg/server/options.go:20]:
```suggestion
// Deprecated: Use WithConfigKey
// WithConfigName option sets the configuration name.
```

File: service/internal/auth/authn.go [hunk 5]
```diff
@@ -197,10 +197,16 @@ func normalizeURL(o string, u *url.URL) string {
 	return ou.String()
 }
 
+// deprecated
```
Comment (jakedoublev) [inline service/internal/auth/authn.go:200]:
```suggestion
// Deprecated: currently still respected within SDK but will be removed
```

**Review by jakedoublev — APPROVED** [review]

**Review by jakedoublev — APPROVED** [review]

**Pull request merged**